### ISSUE TITLE
Flutter, "flutter run -d all" broken for multiple iOS simulators

### ISSUE THREAD

**Issue by ollydixon** [flutter/flutter#61093]
@eseidelGoogle 
Flutter version 17.0.5, stable.

This is something that worked flawlessly last year, has now been broken this entire year. It's causing is a big headache cause many of our apps have user interactions with one another. We also like to develop on multiple screen sizes at the same time.

Right now we are trying to create a new device scaler (cause Flutter doesn't support it) and it's painful to say the least.

**Bug**
Brand new project.
iPhone SE and iPhone 11 simulator open.

"flutter run -d all"

```
The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
techno:my_app olly$ flutter run -d all
Launching lib/main.dart on iPhone 11 in debug mode...
Launching lib/main.dart on iPhone SE (2nd generation) in debug mode...
Running pod install...                                                 ⣻CocoaPods' output:
↳

    [!] Invalid `Podfile` file: File exists @ syserr_fail2_in - .symlinks/plugins/path_provider.

     #  from /Users/olly/Projects/my_app/ios/Podfile:76
     #  -------------------------------------------
     #      symlink = File.join('.symlinks', 'plugins', name)
     >      File.symlink(path, symlink)
     #      pod name, :path => File.join(symlink, 'ios')
     #  -------------------------------------------

    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.9.1/lib/cocoapods-core/podfile.rb:301:in `rescue in block in from_ruby'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.9.1/lib/cocoapods-core/podfile.rb:295:in `block in from_ruby'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.9.1/lib/cocoapods-core/podfile.rb:50:in `instance_eval'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.9.1/lib/cocoapods-core/podfile.rb:50:in `initialize'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.9.1/lib/cocoapods-core/podfile.rb:293:in `new'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.9.1/lib/cocoapods-core/podfile.rb:293:in `from_ruby'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-core-1.9.1/lib/cocoapods-core/podfile.rb:259:in `from_file'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.9.1/lib/cocoapods/config.rb:200:in `podfile'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.9.1/lib/cocoapods/command.rb:150:in `verify_podfile_exists!'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.9.1/lib/cocoapods/command/install.rb:46:in `run'
    /Library/Ruby/Gems/2.6.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.9.1/lib/cocoapods/command.rb:52:in `run'
    /Library/Ruby/Gems/2.6.0/gems/cocoapods-1.9.1/bin/pod:55:in `<top (required)>'
    /usr/local/bin/pod:23:in `load'
    /usr/local/bin/pod:23:in `<main>'

Error running pod install
Error launching application on iPhone SE (2nd generation).
    1.5s
Running Xcode build...        
```

**Comment by VladyslavBondarenko** [comment]
Hi @ollydixon 
I can't reproduce it, app is built and running on both simulators.

```
flutter create someapp
flutter cd someapp
flutter pub get
flutter run -d all --verbose
```

`flutter run -d all --verbose` log
https://gist.github.com/VladyslavBondarenko/2cc01a6d77c7694b41482c92b4f0c210

<details>
 <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, v1.17.5, on Mac OS X 10.15.5 19F101, locale en-GB)
    • Flutter version 1.17.5 at /Users/nevercode/dev/flutter_stable
    • Framework revision 8af6b2f038 (8 days ago), 2020-06-30 12:53:55 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/nevercode/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.1

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.46.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.1

[✓] Connected device (2 available)
    • iPhone 11                  • 5DA82665-75CB-4DCA-B7B1-1172BF571900 • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)
    • iPhone SE (2nd generation) • C63DC848-1CBE-4BB5-A914-92218EA348CE • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)

• No issues found!
```

</details>

Was it indeed new project? It looks like you was running it with `path_provider` and then removed it from `pubspec.yaml`.
Could you provide your `flutter doctor -v` and `flutter run -d all --verbose` log?

**Comment by ollydixon** [comment]
@VladyslavBondarenko brand new project. 

```
[✓] Flutter (Channel stable, v1.17.5, on Mac OS X 10.15.5 19F101, locale en-NO)
    • Flutter version 1.17.5 at /Users/olly/flutter
    • Framework revision 8af6b2f038 (8 days ago), 2020-06-30 12:53:55 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/olly/Library/Android/sdk
    • Platform android-R, build-tools 29.0.3
    • ANDROID_HOME = /Users/olly/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/olly/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.1

[!] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.46.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.1
```

**Comment by VladyslavBondarenko** [comment]
@ollydixon could you attach your `flutter run -d all --verbose` log?

**Comment by ollydixon** [comment]
@VladyslavBondarenko 

Sure, multiple collages are seeing this & our build servers (used for testing) also seeing the same issues. Looks like it's trying to build at the same time instead of one at a time.

https://pastebin.com/5aqxMbmN

**Comment by jmagman** [comment]
@ollydixon
That pastebin doesn't show the `Invalid `Podfile` file: File exists @ syserr_fail2_in - .symlinks/plugins/path_provider.` error, did that go away?

It looks like it worked for the iPhone SE and attached to the observatory:
```
                 {
                   "dataPath" : "\/Users\/olly\/Library\/Developer\/CoreSimulator\/Devices\/AD9C790D-18E2-4723-8695-0ABA0BC49548\/data",
                   "logPath" : "\/Users\/olly\/Library\/Logs\/CoreSimulator\/AD9C790D-18E2-4723-8695-0ABA0BC49548",
                   "udid" : "AD9C790D-18E2-4723-8695-0ABA0BC49548",
                   "isAvailable" : true,
                   "deviceTypeIdentifier" : "com.apple.CoreSimulator.SimDeviceType.iPhone-SE--2nd-generation-",
                   "state" : "Booted",
                   "name" : "iPhone SE (2nd generation)"
                 },
...
[   +3 ms] executing: /usr/bin/xcrun simctl launch AD9C790D-18E2-4723-8695-0ABA0BC49548 com.example.myapp --enable-dart-profiling --enable-checked-mode --verify-entry-points --observatory-port=0
...
[ +207 ms] Observatory URL on device: http://127.0.0.1:56987/Q_0SDu-jRN4=/
```
AD9C790D-18E2-4723-8695-0ABA0BC49548

And then there was a crash in Xcode:
```
[        ] Error output from Xcode build:
           ↳
[        ]     ** BUILD FAILED **
           
           
[        ] Xcode's output:
           ↳
[   +1 ms]     Command line invocation:
                   /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -configuration Debug VERBOSE_SCRIPT_LOGGING=YES -workspace Runner.xcworkspace -scheme Runner BUILD_DIR=/Users/olly/Projects/myapp/build/ios -sdk
iphonesimulator
                   -arch x86_64 SCRIPT_OUTPUT_STREAM_FILE=/var/folders/zs/yncksq793_d828zwbz3b1rjc0000gn/T/flutter_build_log_pipe.HhKYZD/pipe_to_stdout FLUTTER_SUPPRESS_ANALYTICS=true COMPILER_INDEX_STORE_ENABLE=NO
           
               Build settings from command line:
                   ARCHS = x86_64
                   BUILD_DIR = /Users/olly/Projects/myapp/build/ios
                   COMPILER_INDEX_STORE_ENABLE = NO
                   FLUTTER_SUPPRESS_ANALYTICS = true
                   SCRIPT_OUTPUT_STREAM_FILE = /var/folders/zs/yncksq793_d828zwbz3b1rjc0000gn/T/flutter_build_log_pipe.HhKYZD/pipe_to_stdout
                   SDKROOT = iphonesimulator13.5
                   VERBOSE_SCRIPT_LOGGING = YES
           
               note: Using new build system
               note: Building targets in parallel
               error: Build service could not start build operation: unknown error while handling message: Could not acquire lock at path `Path(str:

"/Users/olly/Library/Developer/Xcode/DerivedData/Runner-fpyyzsgpjzigmkeixfoqymdtfvda/Build/Intermediates.noindex/XCBuildData/PIFCache/PROJECT@v11_mod=1594215863.3131104_hash=e24e3cd35d5562926a0d57e26b18a6fa-Project.lock")`
```

Xcode doesn't let you build the same project at the same time with the new build system.  Duplicate of  https://github.com/flutter/flutter/issues/50455#issuecomment-634814531

**Comment by ollydixon** [comment]
@jmagman this is fixed? Why is it closed. Did you test this yourself? If you did you'd see it's broken.

**Comment by ollydixon** [comment]
@VladyslavBondarenko @jmagman can anyone tell me if they have run ```flutter run -d all``` locally with multiple iOS simulators open?

**Comment by VladyslavBondarenko** [comment]
@ollydixon I'm not sure what changed, but yes, now with the same `1.17.5` I'm experiencing https://github.com/flutter/flutter/issues/50455, mentioned above

**Comment by ollydixon** [comment]
@VladyslavBondarenko great, we can reopen this then? Also; I remember that earlier versions of Flutter built once per iOS device. This should be enabled again so we can build across devices again. Right now it's painful to test across multiple iOS devices (A very common use case)
@jmagman can you tell me why the build process was changed? Why was it set to concurrent on iOS?
@eseidelGoogle this is a major issue, many developers are complaining about this in my office.

**Comment by jmagman** [comment]
@ollydixon This is closed because it's a duplicate of #50455, not because it's fixed.  See my comments in https://github.com/flutter/flutter/issues/50455#issuecomment-634814531 for why this may have changed for you (migration to Xcode's "new" build system).
